pkg/apis/monitoring/v1/types.go

```diff
@@ -351,8 +351,8 @@ type PrometheusList struct {
 // +k8s:openapi-gen=true
 type PrometheusSpec struct {
 	CommonPrometheusFields `json:",inline"`
-	// Time duration Prometheus shall retain data for. Default
-	// is '24h', and must match the regular expression `[0-9]+(ms|s|m|h|d|w|y)`
+	// Time duration Prometheus shall retain data for. Default is '24h' if
+	// retentionSize is not set, and must match the regular expression `[0-9]+(ms|s|m|h|d|w|y)`
 	// (milliseconds seconds minutes hours days weeks years).
```

Comment:
This might be a good time to force validation at API level here via regex?

```suggestion
	// +kubebuilder:validation:Pattern=`^[0-9]+(ms|s|m|h|d|w|y)`
	// +optional
```

Reply:
I think that @slashpai has this in her backlog :)